homeassistant/components/group/sensor.py

```diff
@@ -0,0 +1,411 @@
+        vol.Optional(CONF_UNIT_OF_MEASUREMENT): str,
+        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
+        vol.Optional(CONF_STATE_CLASS): STATE_CLASSES_SCHEMA,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the Switch Group platform."""
+    async_add_entities(
+        [
+            SensorGroup(
+                config.get(CONF_UNIQUE_ID),
+                config[CONF_NAME],
+                config[CONF_ENTITIES],
+                config[CONF_ALL],
+                config[CONF_TYPE],
+                config[CONF_ROUND_DIGITS],
+                config.get(CONF_UNIT_OF_MEASUREMENT),
+                config.get(CONF_STATE_CLASS),
+                config.get(CONF_DEVICE_CLASS),
+            )
+        ]
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Initialize Switch Group config entry."""
+    registry = er.async_get(hass)
+    entities = er.async_validate_entity_ids(
+        registry, config_entry.options[CONF_ENTITIES]
+    )
+    async_add_entities(
+        [
+            SensorGroup(
+                config_entry.entry_id,
+                config_entry.title,
+                entities,
+                config_entry.options[CONF_ALL],
+                config_entry.options[CONF_TYPE],
+                int(config_entry.options[CONF_ROUND_DIGITS]),
+                None,
+                None,
+                None,
+            )
+        ]
+    )
+
+
+def calc_min(
+    sensor_values: list[tuple[str, float, State]], round_digits: int
+) -> tuple[dict[str, Any], float]:
+    """Calculate min value."""
+    val: float | None = None
+    entity_id: str | None = None
+    for sensor_id, sensor_value, _ in sensor_values:
+        if val is None or val > sensor_value:
+            entity_id, val = sensor_id, sensor_value
+
+    attributes = {ATTR_MIN_ENTITY_ID: entity_id}
+    if TYPE_CHECKING:
+        assert val is not None
+    return attributes, val
+
+
+def calc_max(
+    sensor_values: list[tuple[str, float, State]], round_digits: int
+) -> tuple[dict[str, Any], float]:
+    """Calculate max value."""
+    val: float | None = None
+    entity_id: str | None = None
+    for sensor_id, sensor_value, _ in sensor_values:
+        if val is None or val < sensor_value:
+            entity_id, val = sensor_id, sensor_value
+
+    attributes = {ATTR_MAX_ENTITY_ID: entity_id}
+    if TYPE_CHECKING:
+        assert val is not None
+    return attributes, val
+
+
+def calc_mean(
+    sensor_values: list[tuple[str, float, State]], round_digits: int
+) -> tuple[dict[str, Any], float]:
+    """Calculate mean value."""
+    result = [sensor_value for _, sensor_value, _ in sensor_values]
```

Comment:
This can be a generator expression.

```suggestion
    result = (sensor_value for _, sensor_value, _ in sensor_values)
```